.github/workflows/test.yml

```diff
@@ -18,12 +18,93 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  RUSTFLAGS: '--cfg async_executor_impl="async-std" --cfg async_channel_impl="async-std"'
+  RUSTFLAGS: '--cfg async_executor_impl="async-std" --cfg async_channel_impl="async-std" --cfg hotshot_example'
   RUST_LOG: info,libp2p=off,node=error
   CARGO_TERM_COLOR: always
 
 jobs:
+  build-test-artifacts:
+    name: Build test artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: rui314/setup-mold@v1
+
+      - uses: actions/checkout@v4
+        with:
+          # By default actions/checkout checks out a merge commit. Check out the PR head instead.
```

Comment:
Why do we want to check out the PR head instead?

Reply:
I might be missing something but I thought we _want_ to merge the PR into main and use that to run the CI so that we are confident things work _after_ a future merge into main.

Reply:
Yea I guess I'm confused about what is happening there.

Reply:
cb8f7e458bfb7482911b1ecbc65478d2fd0d517e